index.js

```diff
@@ -42,21 +56,52 @@ var
     return cWhere (collection, predicates);
   };
 
-var objectList = function objectList (collection) {
-  return {
-    getByKey: function (key) {
-      return getByKey.apply(null, [collection, key]);
-    },
-    whitelist: function (keyWhitelist) {
-      return whitelist.apply(null, [collection, keyWhitelist]);
-    },
-    concat: function () {
-      return concat(collection);
-    },
-    where: function (keyWhitelist) {
-      return where.apply(null, [collection, keyWhitelist]);
-    },
-  };
+objectList = function objectList (options) {
+  var
+    collection,
+    api = {
+      getByKey: function (key) {
+        return getByKey.apply(null, [collection, key]);
+      },
+      whitelist: function (keyWhitelist) {
+        return whitelist.apply(null, [collection, keyWhitelist]);
+      },
+      concat: function () {
+        return concat(collection);
+      },
+      where: function (keyWhitelist) {
+        return where.apply(null, [collection, keyWhitelist]);
+      },
+      add: function (records, cb) {
```

Comment:
This one shouldn't take records or callback. This is the method version which already has access to the collection via closure, and the callback gets hooked up via `.subscribe`.